src/searchers/bfs.rs

```diff
@@ -5,16 +5,21 @@ use crate::{decoders::crack_results::CrackResult, filtration_system::MyResults};
 
 /// Breadth first search is our search algorithm
 /// https://en.wikipedia.org/wiki/Breadth-first_search
-pub fn bfs(input: &str) -> Option<String> {
+pub fn bfs(input: &str, max_depth: u32) -> Option<String> {
     let mut seen_strings = HashSet::new();
     // all strings to search through
     let mut current_strings = vec![input.to_string()];
 
     let mut exit_result: Option<CrackResult> = None;
 
+    // let max_depth = get_config().max_depth;
+
+    let mut curr_depth: u32 = 1; // as we have input string, so we start from 1
+
     // loop through all of the strings in the vec
-    while !current_strings.is_empty() {
+    while !current_strings.is_empty() && curr_depth <= max_depth {
```

Comment:
Suggestion, make a function for `max_depth_checker` and have this:

```rust
fn max_depth_checker(current_depth){
    return curr_depth <= max_depth || !(max_depth < 0)
}
```

That way when max_depth is under 0 (-1) we go infinitely (if I get the logic statements the right way around...)

Reply:
What about using more _rusty_ way?
```rust
Some(max_depth) // if there is limit
None // go infinitely
```

Reply:
hahahahah sure!!!

Reply:
Done